minimum_window_substr.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+Given a string S and a string T, find the minimum window in S which will contain all the characters in T in complexity O(n).
+
+For example,
+S = "ADOBECODEBANC"
+T = "ABC"
+Minimum window is "BANC".
+
+Note:
+If there is no such window in S that covers all characters in T, return the empty string "".
+
+If there are multiple such windows, you are guaranteed that there will always be only one unique minimum window in S.
+*/
+
+#include <unordered_map>
+#include <vector>
+#include <string>
+using namespace std;
+
+class Solution {
+public:
+    string minWindow(string s, string t) {
+        if (s.empty() || t.empty() || s.size() < t.size())
+            return "";
+        unordered_map<char, int> require;
+        for (int i = 0; i < t.size(); i++)
+            require[t[i]]++;
+        vector<int> matchIndices;
+        int foundLeft = -1, foundLen;
+        int matchCount = 0;
+        for (int i = 0; i < s.size(); i++) {
```

Comment:
s.size(), t.size() are used multiple times --> assign to temp vars to save calls.
int sLen = s.size();
int tLen = t.size();